File: SearchingAndSorting/ArraySearchAtmostKDiff/ArraySearchAtmostKDiff.js
function method1(arr,k,target){
    let i,n=arr.length;
    for(i=0;i<n;i++){
        if(arr[i]==target){
            return i;
        }
    }
    return -1;
}

function method2(arr,k,target){
    let i=0,n=arr.length,temp;
    while(i<n){
        if(arr[i]==target)
            return i;
        temp = Math.floor(Math.abs(target-arr[i])/k);
        if(temp==0)
            temp++;
        i+=temp;
    }
}

let arr = [4, 5, 6, 7, 6];
let k = 1;
let target = 6;
console.log(method1(arr,k,target))
console.log(method2(arr,k,target))

File: SearchingAndSorting/ArraySearchAtmostKDiff/ArraySearchAtmostKDiff.go
package main

import (
	"fmt"
	"math"
)

func method1(arr []int, k int, x int) int {
	n := len(arr)
	for i := 0; i < n; {
		if arr[i] == x {
			return i
		}
		diff := int(math.Abs(float64(x - arr[i])))
		ind := diff / k
		if ind == 0 {
			i++
		} else {
			i += ind
		}
	}
	return -1
}

func main() {
	arr := []int{4, 5, 6, 7, 6}
	k := 1
	x := 6
	fmt.Println(method1(arr, k, x))
}
